### PR TITLE
Update Dependencies with Guava

### PR DESCRIPTION
**Pull request by dave2wave**
FYI, https://nvd.nist.gov/vuln/detail/CVE-2023-2976

### CODE REVIEW

File: nb-adapters/adapter-pulsar/pom.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
     </description>
 
     <properties>
-        <pulsar.version>3.2.1</pulsar.version>
+        <pulsar.version>3.0.7</pulsar.version>
```
Comment (msmygit) [inline nb-adapters/adapter-pulsar/pom.xml:37]: @yabinmeng could you take a look at this to ensure if its compatible?

**Review by msmygit — APPROVED** [review]
lgtm. Left 1 commend on pulsar adapter

File: nb-adapters/adapter-pulsar/pom.xml (hunk 1 above)
Comment (dave2wave) [reply nb-adapters/adapter-pulsar/pom.xml:37]: @msmygit In Pulsar branch 3.0.7 is LTS and is aligned for us. The project is not releasing anything on 3.1, 3.2, and 3.3. Version 4.0 was recently released.

Comment (yabinmeng) [reply nb-adapters/adapter-pulsar/pom.xml:37]: It is best to double check with streaming team. I'm not really sure

Comment (dave2wave) [reply nb-adapters/adapter-pulsar/pom.xml:37]: They have been asked. We'll take this change now and can adjust at their direction later.

**Pull request merged**